pkg/services/ngalert/state/historian/annotation.go

```diff
@@ -40,12 +40,17 @@ func NewAnnotationBackend(annotations annotations.Repository, dashboards dashboa
 }
 
 // RecordStates writes a number of state transitions for a given rule to state history.
-func (h *AnnotationBackend) RecordStatesAsync(ctx context.Context, rule *ngmodels.AlertRule, states []state.StateTransition) {
+func (h *AnnotationBackend) RecordStatesAsync(ctx context.Context, rule *ngmodels.AlertRule, states []state.StateTransition) <-chan error {
 	logger := h.log.FromContext(ctx)
 	// Build annotations before starting goroutine, to make sure all data is copied and won't mutate underneath us.
 	annotations := h.buildAnnotations(rule, states, logger)
 	panel := parsePanelKey(rule, logger)
-	go h.recordAnnotationsSync(ctx, panel, annotations, logger)
+	errCh := make(chan error, 1)
+	go func() {
+		defer close(errCh)
+		errCh <- h.recordAnnotationsSync(ctx, panel, annotations, logger)
+	}()
+	return errCh
```

Comment:
We're wanting to eventually bump this up to the state manager in a future PR, correct? No feedback here, just trying to confirm my understanding.

Reply:
Hmm, I'm not sure I understand what you mean about the state manager? This is just about providing a cancel signal to avoid goroutine leaks in `context.Context`.

Reply:
Would you be happy for me to merge this to unblock @alexmobo? I'm happy to keep iterating on it 🙂

Reply:
I'm going to merge, let's keep iterating on it though 🙂

Reply:
Yeah, happy to merge - sorry, I thought I hit approve with this comment, but it seems I didn't :smile:

Reply:
Anyway, LGTM!